src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -530,7 +530,7 @@ private void ReportDiagnosticsIfObsoleteCore(Binder binder, SyntaxNode syntax, B
         /// </summary>
         public bool Is(TypeParameterSymbol other)
         {
-            return NullableAnnotation.IsOblivious() && ((object)DefaultType == other) &&
+            return !NullableAnnotation.IsAnnotated() && ((object)DefaultType == other) &&
```

Comment:
This is breaking other things. Closing the PR for now. If I could get a correct fix, I'll open a new PR.